kobo/shortcuts.py

```diff
@@ -284,6 +284,14 @@ def run(cmd, show_cmd=False, stdout=False, logfile=None, can_fail=False, workdir
         cmd = " ".join((shlex_quote(i) for i in cmd))
 
     universal_newlines = kwargs.get('universal_newlines', False)
+    # new args added in py3 for Popen
+    text = kwargs.get('text', False)
+    encoding = kwargs.get('encoding', None)
+    errors = kwargs.get('errors', None)
+
+    # any of these args is passed, text mode will be enabled.
+    is_text_mode = any([universal_newlines, text, encoding, errors])
+    encoding_method = encoding or 'utf-8'
```

Comment:
To me it'd be cleaner to just reuse encoding variable, like:

```suggestion
    encoding = encoding or 'utf-8'
```

The `encoding_method` name is a little awkward, it's unclear that `encoding` and `encoding_method` really mean the same thing except that one of them is the raw value passed by the caller and the other is the value with some default possibly applied.  How about just applying the default to `encoding` directly?

Reply:
yeah, makes sense.